src/fluree/db/ledger/indexing.clj

```diff
@@ -398,53 +400,134 @@
+          (if current?
+            (do
+              (session/release-indexing-lock! session indexed-block)
+              (assoc block-map :db-after indexed-db))
+            (throw (ex-info (str "Recently re-indexed DB on disk is not current after 2 second pause. "
+                                 "Pending new block: " block ", latest block on disk: " (:block latest-db) ".")
+                            {:status 500 :error :db/unexpected-error}))))))))
+
+
+(defn merge-new-index
+  ([session block-map reindexed-db]
+   (merge-new-index session block-map reindexed-db 0))
+  ([session {:keys [flakes block] :as block-map} reindexed-db retries]
+   (log/warn "MERGING NEW INDEX: " {:old (:db-after block-map)
+                                    :new reindexed-db})
    (go-try
-     (let [{:keys [block network]} db
-           last-index (session/indexed session)]
+     (let [max-retries           20
+           pause-before-retry-ms 100
+           indexed-block         (:block reindexed-db)
+           current?              (= indexed-block (dec block))]
        (cond
-         (and last-index (<= block last-index))
-         (do
-           (log/info "Index called on DB but last index isn't older."
-                     {:last-index last-index :block block :db (pr-str db) :session (pr-str session)})
-           false)
-
-         (session/acquire-indexing-lock! session block)
-         (let [updated-db (<? (index db opts))
-               group      (-> updated-db :conn :group)]
-           ;; write out index point
-           (<? (txproto/write-index-point-async group updated-db))
-           (session/clear-db! session)                      ;; clear db cache to force reload
-           (session/release-indexing-lock! session block)   ;; free up our lock
-           (<? (index* session opts))                       ;; run a new index check in case we need to start another one immediately
-           true)
+         ;; everything is good, re-associate db-after with reindexed db
+         current?
+         (let [updated-db (<? (dbproto/-with reindexed-db block flakes))]
+           (session/release-indexing-lock! session (:block updated-db))
+           (assoc block-map :db-after updated-db))
+
+         (= retries max-retries)
+         (throw (ex-info (str "Recently re-indexed DB on disk is not current after " max-retries " retries. "
+                              "Pending new block: " block ", latest block on disk: " (:block reindexed-db) ".")
+                         {:status 500 :error :db/unexpected-error}))
 
          :else
-         (do
-           (log/warn "Indexing process failed to obtain index lock. Extremely Unusual."
-                     {:network network :db (pr-str db) :block block :indexing? (session/indexing? session)})
-           false))))))
+         (let [_ (log/warn "###### MERGING NEW INDEX WAIT - retries: " retries
+                           "indexed-block: " indexed-block "current-block: " block)
```

Comment:
Another log message that you may or may not want to keep (at `warn` level and/or with shouty all caps)

Reply:
Fixed logging I left in by mistake.